### PR TITLE
Change enterprise search connector replica setting from 0-3 to 0-1

### PR DESCRIPTION
**Pull request by dakrone**
This updates the `index.auto_expand_replicas` setting for the enterprise search indices to use a smaller number of expanded replicas.

### CODE REVIEW

**Comment by carlosdelest** [comment]
I've been checking the reasoning on Ent Search side, and came just with [this Slack thread](https://elastic.slack.com/archives/CBSD8CFQD/p1559696713033400).

I'm OK with the change but would like some context for this change. @dakrone can you please elaborate? 🙏

**Comment by sphilipse** [comment]
Hi @carlosdelest, this setting has apparently been causing issues for Stateless Elasticsearch, which doesn't have a concept of replicas. While Elasticsearch will eventually handle this behind the scenes, for now it's causing projects to go yellow (because these indices don't have replicas). As far as I can tell there's no real reason why we should have this set to anything other than 0-1, which appears to be the default for system indices.

**Comment by carlosdelest** [comment]
Got it - so if this does apply to just connector management indices, I see no problem in doing this. Thanks for the context!

**Review by carlosdelest — APPROVED** [review]

**Review by kderusso — APPROVED** [review]
This is fine, but note that if we want to keep this at 3 shards for stateful, we can close this PR and open a serverless PR.

File: x-pack/plugin/core/template-resources/src/main/resources/entsearch/connector/elastic-connectors-settings.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "settings": {
       "hidden": true,
       "number_of_shards": "1",
-      "auto_expand_replicas": "0-3",
+      "auto_expand_replicas": "0-1",
```
[the remark this answers is not in the document]
Comment (dakrone) [reply x-pack/plugin/core/template-resources/src/main/resources/entsearch/connector/elastic-connectors-settings.json:6]: Yes, good idea, I pushed a commit to do this.

**Review by sphilipse — APPROVED** [review]

**Pull request merged**